Exiled.CustomItems/API/Features/CustomItem.cs

```diff
@@ -710,8 +710,13 @@ private void OnInternalPickingUp(PickingUpItemEventArgs ev)
         private void OnInternalChanging(ChangingItemEventArgs ev)
         {
             if (!Check(ev.NewItem))
+            {
+                Exiled.API.Extensions.MirrorExtensions.ResyncSyncVar(ev.Player.ReferenceHub.networkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_myNickSync));
                 return;
+            }
 
+            foreach (Player player in Player.Get(RoleType.Spectator))
```

Comment:
Shouldn't the nickname be synced only if the item is "hacky"?

Reply:
quite right